sympy/series/formal.py

```diff
@@ -589,19 +589,26 @@ def _transform_explike_DE(DE, g, x, order, syms):
         coeff = DE.coeff(Derivative(g(x), x, i))
         coeff = (coeff / highest_coeff).expand().collect(x)
         for t in Add.make_args(coeff):
-            if t.has(x):
-                eq.append(t)
-    sol = solve(eq, syms, dict=True)
-    if sol:
-        DE = DE.subs(sol[0])
-        DE = DE.factor().as_coeff_mul(Derivative)[1][0]
-        DE = DE.collect(Derivative(g(x)))
+            eq.append(t)
+    temp = []
+    for e in eq:
+        if e.has(x):
+            break
+        temp.append(e)
+    else:
+        eq = temp
+    if eq:
+        sol = dict(zip(syms, (i for s in linsolve(eq, list(syms)) for i in s)))
+        if sol:
+            DE = DE.subs(sol)
+            DE = DE.factor().as_coeff_mul(Derivative)[1][0]
+            DE = DE.collect(Derivative(g(x)))
```

Comment:
This is used in the case of sin(x)*cos(x), which is in the test cases.

On 8 August 2015 at 01:11, AMiT Kumar <notifications@github.com> wrote:

> In sympy/series/formal.py
> <https://github.com/sympy/sympy/pull/9789#discussion_r36553895>:
>
> > -        DE = DE.factor().as_coeff_mul(Derivative)[1][0]
> > -        DE = DE.collect(Derivative(g(x)))
> > +            eq.append(t)
> > +    temp = []
> > +    for e in eq:
> > +        if e.has(x):
> > +            break
> > +        temp.append(e)
> > +    else:
> > +        eq = temp
> > +    if eq:
> > +        sol = dict(zip(syms, (i for s in linsolve(eq, list(syms)) for i in s)))
> > +        if sol:
> > +            DE = DE.subs(sol)
> > +            DE = DE.factor().as_coeff_mul(Derivative)[1][0]
> > +            DE = DE.collect(Derivative(g(x)))
>
> The above three lines are untested now.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/sympy/sympy/pull/9789/files#r36553895>.
>



-- 
Regards
*Sartaj Singh*

*Mathematics and Computing*,
Indian Institute of Technology,
Varanasi - 221 005 INDIA

E-mail: singhsartaj94@gmail.com, *sartaj.singh.apm13@itbhu.ac.in
<sartaj.singh.apm13@itbhu.ac.in>*